rclcpp/test/resources/test_node_parameters/wildcards.yaml

```diff
@@ -0,0 +1,36 @@
+/**:
+  ros__parameters:
+    full_wild: "full_wild" 
+
+/*:
+  node:
+    ros__parameters:
+      namespace_wild: "namespace_wild" 
+
+ns:
+  "*":
```

Comment:
Even if * without double quotes in YAML is invalid, I still think the node wildcards without (") here should be supported, just like the namespace, such as (/*). 
```
ns:
  /*:           # "*" can be also supported, but I think users like this style.
    ros__parameters:
      node_wild_in_ns: "node_wild_in_ns" 
```

It should be discussed with the maintainer. [rcl_yaml_param_parser](https://github.com/ros2/rcl/blob/911088d8cb6ecba14290c399cc65f86adc3e3bf7/rcl_yaml_param_parser/src/parser.c#L205-L223) should be updated if the above suggestion is accepted.

Reply:
Supporting both `/*` and `"*"` makes sense to me :+1:

Reply:
I take back my previous comment. Name tokens separated by a YAML colon (`:`) should be treated as if they are separated with a slash (`/`). Therefore, the example:

```yaml
ns:
  /*:
    ros__parameters:
      ...
```

Should resolve to the name `/ns//*`, which is invalid. The correct way to write it would be:

```yaml
/ns/*:
  ros__parameters:
    ...
```